authors/apps/comments/tests/tests.py

```diff
@@ -0,0 +1,261 @@
+
+    def test_get_all_comments(self):
+        """
+        Test that all comments are retrieved on sending a GET request
+        """
+        res = self.client.get(
+            reverse(
+                'comments:create-list',
+                kwargs={
+                    "slug": self.article.slug
+                }
+            ),
+            format="json"
+        )
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+    def test_get_specific_comment(self):
+        """
+        Test that a specific comment is retrieved on sending a GET request
+        """
+        res = self.client.get(
+            reverse(
+                'comments:details',
+                kwargs={
+                    "slug": self.article.slug,
+                    'pk': self.comment_id
+                }
+            ),
+            format="json"
+        )
+
+        self.assertEqual(self.comment.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+    def test_error_on_invalid_data(self):
+        """
+        Test that an error is displayed when invalid values are supplied
+        """
+        res1 = self.client.post(
+            reverse(
+                'comments:create-list',
+                kwargs={
+                    "slug": self.article.slug
+                }
+            ),
+            data={
+                "comment": {
+                    "body": "      "
+                }
+            },
+            format="json"
+        )
+
+        res2 = self.client.post(
+            reverse(
+                'comments:create-list',
+                kwargs={
+                    "slug": self.article.slug
+                }
+            ),
+            data={
+                "comment": {
+                    "body": True
+                }
+            },
+            format="json"
+        )
+
+        self.assertEqual(res1.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(res2.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_comment(self):
+        """
+        Test that a comment is updated successfully
+        """
+
+        res = self.client.put(
+            reverse(
+                'comments:details',
+                kwargs={
+                    "slug": self.article.slug,
+                    'pk': self.comment_id
+                }
+            ),
+            data={
+                "comment": {
+                    "body": "This is an updated comment"
+                }
+            },
+            format="json"
+        )
+
+        self.assertEqual(self.comment.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertNotEqual(
+            self.comment.data.get('comment'),
+            res.data.get('comment')
+        )
+
+    def test_delete_comment(self):
+        """
+        Test that comment can be deleted
+        """
+        create_res = self.client.post(
```

Comment:
For consistency purposes, wouldn't have been wise to call the `self.comment` at line `76`, instead of creating another here?

Reply:
Done